Bio/SeqIO/GckIO.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2019 Damien Goutte-Gattat.  All rights reserved.
+#
+# This file is part of the Biopython distribution and governed by your
+# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
+# Please see the LICENSE file that should have been included as part of this
+# package.
+
+"""Bio.SeqIO support for the "gck" file format.
+
+The GCK binary format is generated by the Gene Construction Kit software
+from Textco BioSoftware, Inc.
+"""
+
+from struct import unpack
+
+from Bio import Alphabet
+from Bio.Seq import Seq
+from Bio.SeqFeature import SeqFeature, FeatureLocation
+from Bio.SeqRecord import SeqRecord
+
+
+def _read(handle, length):
+    """Read the specified number of bytes from the given handle."""
+    data = handle.read(length)
+    if len(data) < length:
+        raise ValueError("Cannot read {} bytes from handle".format(length))
+    return data
+
+
+def _read_packet(handle):
+    """Read a length-prefixed packet.
+
+    Parts of a GCK file are made of "packets" comprising of 4 bytes
+    giving the packet's size, followed by the packet's data.
+
+    There is no type tag. The type of a packet, and thus the type of data
+    it contains, is solely indicated by the position of the packet within
+    the GCK file.
+    """
+    length = _read(handle, 4)
+    length = unpack('>I', length)[0]
+    data = _read(handle, length)
+    return (data, length)
+
+
+def _read_pstring(handle):
+    """Read a Pascal string.
+
+    A Pascal string is one byte for length followed by the actual string.
+    """
+    length = _read(handle, 1)
+    length = unpack('>B', length)[0]
+    data = _read(handle, length).decode('ASCII')
+    return data
+
+
+def _read_p4string(handle):
+    """Read a 32-bit Pascal string.
+
+    Similar to a Pascal string but length is encoded on 4 bytes.
+    """
+    length = _read(handle, 4)
+    length = unpack('>I', length)[0]
+    data = _read(handle, length).decode('ASCII')
+    return data
+
+
+def GckIterator(handle):
+    """Parse a GCK file and return a SeqRecord object."""
```

Comment:
Is there only ever one record per file?

Reply:
In all the GCK, SnapGenes, or Xdna files that I have ever seen, yes. Those formats seem strictly mono-sequence.

Reply:
Could you put a comment about that in the docstring then (as done for one of the other new formats). Thanks